core/runtime/execute_engine.cpp

```diff
@@ -140,15 +140,17 @@ void setup_input_tensors(
     } else {
       at::Tensor contig_input = inputs[i].view(shape).contiguous();
       formatted_inputs.emplace_back(std::move(contig_input));
-
+      bool need_cudagraphs_record = cudagraphs_enabled and
+          (not compiled_engine->runtime_states.old_cudagraphs or shape_changed or
+           compiled_engine->runtime_states.context_changed);
       if (need_cudagraphs_record) {
         // Create a new persistent input buffer
         compiled_engine->input_buffers[i] = std::move(formatted_inputs.back().clone());
       }
-
-      TORCHTRT_CHECK(
-          compiled_engine->exec_ctx->setInputShape(name.c_str(), dims), "Error while setting the input shape");
-
+      if (shape_changed or compiled_engine->allocated_outputs.size() == 0) {
```

Comment:
What is the allocated_outputs size test? Just that we havent allocated the outputs yet?